connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConnectorTest.java

```diff
@@ -672,6 +452,27 @@ protected void assertInitializedMetric(WorkerConnector workerConnector, String e
         assertEquals(VERSION, version);
     }
 
+    @SuppressWarnings("unchecked")
+    private Callback<TargetState> mockCallback() {
+        return mock(Callback.class);
+    }
+
+    private void verifyCleanInitialize() {
+        verify(connector).version();
+        if (connector instanceof SourceConnector) {
+            verify(offsetStore).start();
+            verify(connector).initialize(any(SourceConnectorContext.class));
+        } else {
+            verify(connector).initialize(any(SinkConnectorContext.class));
+        }
+    }
+
+    private void verifyCleanShutdown() {
+        verify(ctx).close();
+        verify(offsetStorageReader).close();
```

Comment:
This is a little unrealistic since we should only be supplying non-null offset readers for source connectors. Can we correct that logic while we're in the neighborhood?

Reply:
Good catch, thanks! I just translated the tests as is and hadn't noticed that we were passing non-null offset storage reader and offset backing stores instances even for sink connectors in these tests (differing in behavior with how the actual `Worker` instantiates `WorkerConnector`s).